### PR TITLE
Fixed configs (env isolation)

### PR DESCRIPTION
**Pull request by mtitov**

### CODE REVIEW

**Comment by mtitov** [comment]
@vrpascuzzi can you please make a test on Arcticus with this branch and confirm the correctness of separation of loaded modules (between bootstrapping and launch method)

**Review by andre-merzky — APPROVED** [review]
Thanks!  LGTM

**Comment by vrpascuzzi** [comment]
> @vrpascuzzi can you please make a test on Arcticus with this branch and confirm the correctness of separation of loaded modules (between bootstrapping and launch method)

For sure. Will get back to you momentarily.

File: src/radical/pilot/configs/resource_anl.json [hunk 1]
```diff
@@ -79,14 +81,20 @@
...
         "agent_spawner"               : "POPEN",
-        "agent_launch_method"         : "SSH",
-        "task_launch_method"          : "MPIRUN",
-        "mpi_launch_method"           : "MPIRUN",
+        "launch_methods"              : {
+                                         "order" : ["MPIRUN", "SSH"],
+                                         "MPIRUN": {
+                                             "pre_exec_cached": [
+                                                 "module use  /soft/modulefiles",
+                                                 "module load openmpi/4.1.1-gcc"
```
Comment (vrpascuzzi) [inline src/radical/pilot/configs/resource_anl.json:89]: Just for future reference, MPICH is expected to be the primary MPI used on Aurora.

**Comment by vrpascuzzi** [comment]
@mtitov Quick test didn't work:
```
================================================================================
 Getting Started (RP version 1.10.0)                                            
================================================================================

new session: [rp.session.jlselogin5.ac.vpascuzzi.018961.0002]                  \
database   : [mongodb://pascuzzi:****@10.17.6.81/lavinlie]                    ok
create pilot manager                                                          ok
create task manager                                                           ok

--------------------------------------------------------------------------------
submit pilots                                                                   

submit 1 pilot(s)caught Exception: pilot size must be defined
--------------
RADICAL Utils -- Stacktrace [51338] [MainThread]

ac.vpas+  51338  48387  4 21:27 pts/22   00:00:00  |           \_ python /home/ac.vpascuzzi/radical/radical.pilot/examples/00_getting_started.py anl.iris
Traceback (most recent call last):
File "/home/ac.vpascuzzi/radical/radical.pilot/examples/00_getting_started.py", line 67, in <module>
pilot = pmgr.submit_pilots(pdesc)
File "/home/ac.vpascuzzi/radical/ve/lib/python3.7/site-packages/radical/pilot/pilot_manager.py", line 588, in submit_pilots
raise ValueError('pilot size must be defined')
ValueError: pilot size must be defined

--------------
<...>
```

**Comment by vrpascuzzi** [comment]
I've made some other changes for Arcticus. I will try applying just your patch to `resource_anl.json` -- since that's the only file changed in this PR that matters for Arcticus -- and test again.

**Review by vrpascuzzi — CHANGES_REQUESTED** [review]
Can remove l78 in [resource_anl.json](https://github.com/radical-cybertools/radical.pilot/pull/2476/files#diff-18230d11b6a63166bf13cce24180e421a858aa810a6997216791a4a5cf815f69R78).

**Comment by andre-merzky** [comment]
> Can remove l78 in [resource_anl.json](https://github.com/radical-cybertools/radical.pilot/pull/2476/files#diff-18230d11b6a63166bf13cce24180e421a858aa810a6997216791a4a5cf815f69R78).

I commented that out, assuming it it not needed due to the manual tunnel setup.  If that is correct, then yes, that line can be removed.

> `raise ValueError('pilot size must be defined')`

This seems to be an error in the pilot description - I would assume that `cores` is not set?

**Comment by vrpascuzzi** [comment]
Apologies for the delay -- all nodes allocated...

**Comment by vrpascuzzi** [comment]
I confirm using my local changes works:
```
<...>
submit 2 pilot(s)
        pilot.0000   anl.arcticus             96 cores       2 gpus
        pilot.0001   anl.arcticus             96 cores       2 gpus           ok

--------------------------------------------------------------------------------
submit 16 tasks                                                                 

create: ########################################################################
submit: ########################################################################
wait  : ########################################################################
	DONE      :    16
<...>
```

 Unfortunately, need to run now. Will follow-up in tomorrow.

**Comment by mtitov** [comment]
@vrpascuzzi just to confirm that with "follow-up" you meant you approve current changes, but ready to follow new requests if any, right? and should we remove line 78 completely (or it works for test purposes)?

p.s. CI test analytics is failed due to new major version of `pymongo`, thus RCT will use any previous version - this is about to be fixed

**Comment by vrpascuzzi** [comment]
@mtitov Apologies — distractions all day.

I hope to be back online this evening at some point. But if this needs prompt merging, I have no strong objection. I don’t think an issue related to arcticus support is super urgent to address right now.

**Review by andre-merzky — APPROVED** [review]
Thanks, LGTM

**Pull request merged**